crates/astria-composer/src/searcher/executor.rs

```diff
@@ -120,178 +139,266 @@ impl Executor {
+        .on_retry(
+            |attempt,
+             next_delay: Option<Duration>,
+             err: &sequencer_client::extension_trait::Error| {
+                let wait_duration = next_delay
+                    .map(humantime::format_duration)
+                    .map(tracing::field::display);
+                let err = err.clone();
+                let span = warn_span!(parent: span.clone(), "report attempt failure");
+                async move {
+                    let error = &err as &(dyn std::error::Error + 'static);
+                    warn!(
+                        attempt,
+                        wait_duration,
+                        error,
+                        "failed getting latest nonce from sequencer; retrying after backoff",
+                    );
+                }
+                .instrument(span)
+            },
+        );
+    tryhard::retry_fn(|| {
+        let client = client.clone();
+        let span = info_span!(parent: span.clone(), "attempt get nonce");
+        async move { client.get_latest_nonce(address).await.map(|rsp| rsp.nonce) }.instrument(span)
+    })
+    .with_config(retry_config)
+    .await
+    .wrap_err("failed getting latest nonce from sequencer after 1024 attempts")
+}
 
-        self.nonce = Some(nonce_response.nonce);
-        info!(
-            nonce_response.nonce,
-            "retrieved initial nonce from sequencer successfully"
+/// Queries the sequencer for the latest nonce with an exponential backoff
+#[instrument(
+    name = "submit signed transaction",
+    skip_all,
+    fields(
+        nonce = tx.unsigned_transaction().nonce,
+        transaction.hash = hex::encode(tx.to_raw().encode_to_vec()),
+    )
+)]
+async fn submit_tx(
+    client: sequencer_client::HttpClient,
+    tx: SignedTransaction,
+) -> eyre::Result<tx_sync::Response> {
+    debug!("submitting signed transaction to sequencer");
+    let span = Span::current();
+    let retry_config = tryhard::RetryFutureConfig::new(1024)
+        .exponential_backoff(Duration::from_millis(200))
+        .max_delay(Duration::from_secs(60))
+        .on_retry(
+            |attempt,
+             next_delay: Option<Duration>,
+             err: &sequencer_client::extension_trait::Error| {
+                let wait_duration = next_delay
+                    .map(humantime::format_duration)
+                    .map(tracing::field::display);
+                let err = err.clone();
+                let span = warn_span!(parent: span.clone(), "report attempt failure");
+                async move {
+                    let error = &err as &(dyn std::error::Error + 'static);
+                    warn!(
+                        attempt,
+                        wait_duration,
+                        error,
+                        "failed getting latest nonce from sequencer; retrying after backoff",
+                    );
+                }
+                .instrument(span)
+            },
         );
```

Comment:
`retry_config` appears to be the same as above - can you make a function for it

Reply:
As you flagged in another comment: the warn message was wrong, so now the config are almost the same, but stil different.